src/lazy/text/raw/reader.rs

```diff
@@ -97,6 +98,34 @@ mod tests {
 
     use super::*;
 
+    struct TestReader<'data> {
+        allocator: BumpAllocator,
+        reader: LazyRawTextReader_1_0<'data>,
+    }
```

Comment:
🗺️ A consequence of expression caching is that using the raw reader requires the application to supply a `BumpAllocator` to each call to `next()`. Since this is not an API many people will interact with--most won't need to and it will be behind a feature flag--I'm ok with that tradeoff.

Reply:
This is fine, but is there any particular reason it can't be stored in the reader?

Reply:
`LazyRawReader::next()` mutably borrows the raw reader for `'top`, and the expanding reader needs access to the allocator after calling `LazyRawReader::next()` which would require another borrow.

An alternative to passing the allocator into `next()` is having `next()` return a reference to the allocator with each value. That's possible (and maybe preferable?) but I'd like to defer implementing that until after this PR.